### PR TITLE
Startup time improvement and Dynamic Shortcuts

### LINKED ISSUE #1314: Adding shortcuts to Android
Feature Request:

I want to introduce a feature of adding static shorts to the app for fragments/activities for quick access to users like:

Implementing this feature will increase the User Experience (UX) of the app and make parts of app much more accessible.

Feature description
App shortcuts were introduced in API 25. for the users to do their common tasks quickly.

Screenshots:
static2

### PR DESCRIPTION
**Pull request by nirmalhk7**
- Fixes #1314 
- I've also removed the SetContentView() in SplashScreen, which is unnecessary as the SplashScreen already loads the logo from the style.xml, thus improving startup time.

Please Add Screenshots If there are any UI changes.

untitled

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `MifosStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/online/DashboardActivity.java [hunk 1]
```diff
@@ -122,6 +132,66 @@ public void onBackStackChanged() {
...
+            shortcutList.add(
+                    new ShortcutInfo.Builder(this, getString(R.string.settings))
+                            .setShortLabel(getString(R.string.settings))
+                            .setLongLabel(getString(R.string.settings))
+                            .setIcon(Icon.createWithResource(this,
+                                    R.drawable.ic_settings))
+                            .setIntent(intent.setAction(getString(R.string.settings)))
+                            .build()
+            );
+            shortcutManager.setDynamicShortcuts(shortcutList);
```
Comment (Gelassen) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/online/DashboardActivity.java]: Don't you want to do this logic in cycle? Code is repeat five times, instead you could move this strings and drawable consts into two arrays and iterate over it by creating shortcut and adding it into the final collection.

**Comment by robustTechie** [comment]
Are all the shortcuts working?

**Comment by nirmalhk7** [comment]
> Are all the shortcuts working?

@robustTechie Yes, they are.

**Comment by robustTechie** [comment]
Also, add screen recoding for that, so it would be better for the reviewer to see the changes. :)

**Comment by RajaVamsi11** [comment]
@nirmalhk7 Please squash your commits into one.

**Comment by nirmalhk7** [comment]
> @nirmalhk7 Please squash your commits into one.

I have a merge commit between two of my other commits, and Im not quite sure how to rebase them now because I really dont want to mess Git's history. Do you have any suggestions on what to do?

**Comment by RajaVamsi11** [comment]
@nirmalhk7 Why are other contributors commits are showing in your commit?

**Comment by nirmalhk7** [comment]
> @nirmalhk7 Why are other contributors commits are showing in your commit?

@RajaVamsi11 I'm not quite sure, and that is why I dont want to mess around. I dont mind redoing my work again, but I would mind if someone else had to redo their work because I wasn't careful.

**Comment by RajaVamsi11** [comment]
The commits that are showing up in your commit list are already merged in the master branch. You do not worry about their commits as they are already merged.

**Comment by nirmalhk7** [comment]
> The commits that are showing up in your commit list are already merged in the master branch. You do not worry about their commits as they are already merged.

Okay, so how do I squash then?

**Comment by RajaVamsi11** [comment]
You can follow this one https://www.internalpointers.com/post/squash-commits-into-one-git

**Comment by AbhilashG97** [comment]
Hello @nirmalhk7,

Can you please squash your commits?

I'm closing your pull request for now. Please re-open this pull request when you have made the suggested changes.

Thank you for your contribution.

**Pull request closed without merging**